### DISCUSSION TITLE
蓝盾agent失败之后流水线显示为构建成功

### DISCUSSION THREAD

**Discussion by QiLong-616** [Tencent/bk-ci#4299] [Ideas]
![image](https://user-images.githubusercontent.com/78772668/120260032-a1935580-c2c7-11eb-813f-3e39285493a9.png)
![image](https://user-images.githubusercontent.com/78772668/120260977-89bcd100-c2c9-11eb-88f3-21bc07f67d6b.png)
![image](https://user-images.githubusercontent.com/78772668/120261033-a78a3600-c2c9-11eb-8593-0270f1a5e716.png)

  在这个agent失败之后能不能将流水线的状态设为失败，因为在外面看到是成功的就不会太关注这个流水线了，当需要用的时候点进去查看，才发现原来是失败了，又得重新构建这样会浪费很多的时间，能否解决一下

**Comment by zanyzhao** [comment]
1. 我看子流水线是的package步骤是黄色的，是被终止了吗？
2. 你是如何判断子流水线的构建是失败的？

**Reply by QiLong-616** [reply]
![image](https://user-images.githubusercontent.com/78772668/120261058-b7097f00-c2c9-11eb-9191-3dfb332ef50d.png)
点开这个子流水线里面显示的是agent心跳超时，但是它整体的表现是绿色的，如果正常取消的话也是红色失败的啊

**Reply by irwinsun** [reply]
这个问题是因为 触达了插件的控制流程逻辑： 失败继续 引起的。
不过这个真的是个好问题：
因为构建机已经挂掉了无法恢复,  所以即使所在插件 设置了 “失败继续”， 也无法继续运行下去了。 理应该为失败！

**Comment by zanyzhao** [comment]
已转issue #4301
